pkg/query-service/utils/labels/labels.go

```diff
@@ -106,7 +105,7 @@ func (ls Labels) HashForLabels(names ...string) uint64 {
 			}
 		}
 	}
-	return xxhash.Sum64(b)
+	return xxhash.Sum64(b), b
```

Comment:
@srikanthccv is the returned `b` used anywhere? It might breaking changes if used as it is modified.

Reply:
Good question, I don't know much about this file. What I remember is this whole file was actually taken from an upstream file. I tried to just copy. Let me check.

Reply:
Don't particularly remember where and which version this was copied from. Let me ask @mindhash since he did this for some reason but I believe the latest one is this https://github.com/prometheus/prometheus/blob/main/model/labels/labels.go.

Reply:
I looked into the rules engine. HashForLabels is not used by anything in our usecase.  

The utils were copied over from prom to reduce direct references to prom package. At the time, we had thought of completely removing prom rules, hence a copy of was made.

A rules engine relies on BaseLables interface which has prom implementation and our own (query-service) implementation. Both these implementations are identical.

Reply:
@mindhash I see that the Prometheus implementation now accepts the byte array `b []byte` but immediately does `b = b[:0]` which I imagine they are doing to reduce the allocations and got some perf improvements. Should just copy the upstream code once again?

Reply:
We are not using this function.(Could not find any refs in the code) 
So I am not sure if we need to pull upstream.

Reply:
I have copied the upstream implementation, so this should be fine whether or not it is used.